targets/core

```diff
@@ -106,18 +106,62 @@ EOF
     apt-get -y dist-upgrade
 fi
 
+# Update the portage tree for Gentoo, if at least one day has passed
+if [ "${DISTROAKA:-"$DISTRO"}" = 'gentoo' ]; then
+    # Get the time of the last sync
+    LAST_SYNC=$(cut -f1 -d' ' /usr/portage/metadata/timestamp.x)
+    # Add a day to it, in seconds
+    LAST_SYNC=$((LAST_SYNC + 86400))
+    # Get the current epoch time in seconds
+    TIME_SEC=$(date +%s)
+    # If at least a day has passed, do the sync
+    if [ "$TIME_SEC" -gt "$LAST_SYNC" ]; then
+        emerge --sync
+    fi
+
+    # Make sure that the system is up to date
+    emerge world -Duv --newuse
+fi
+
 # On release upgrade, keyboard-configuration might be reconfigured.
 fixkeyboardmode
 
+# 
+if [ "${DISTROAKA:-"$DISTRO"}" = 'gentoo' ]; then
+    # Install gentoolkit for euse, so we can set use flags
+    # Install cpuid2cpuflags so we can easily set reasonable CFLAGS
+    install gentoolkit cpuid2cpuflags
+    if [ "$ARCH" = 'x86' -o "$ARCH" = 'amd64' ]; then
```

Comment:
this may need to be changed depending on whether you want ARCH to be x86 or i686 (same with arm -- armhf or armv7-hardf or whatever it is).